src/frontends/pytorch/src/input_model.cpp

```diff
@@ -0,0 +1,127 @@
+}
+
+Place::Ptr InputModel::get_place_by_tensor_name(const std::string& tensor_name) const {
+    auto place_it = m_name_to_place.find(tensor_name);
+    FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for provided tensor name.");
+    return place_it->second;
+}
+
+void InputModel::set_partial_shape(const Place::Ptr& place, const ov::PartialShape& shape) {
+    FRONT_END_GENERAL_CHECK(place->is_input(), "Setting partial shape is supported for input only.");
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
+    auto it = m_descriptors.find(pytorch_place->get_tensor_index());
+    if (it != m_descriptors.end()) {
+        it->second.m_pshape = shape;
+    }
+}
+
+ov::PartialShape InputModel::get_partial_shape(const Place::Ptr& place) const {
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
+    auto it = m_descriptors.find(pytorch_place->get_tensor_index());
+    if (it != m_descriptors.end()) {
+        return it->second.m_pshape;
+    }
+    return PartialShape::dynamic();
+}
+
+void InputModel::set_element_type(const Place::Ptr& place, const ov::element::Type& type) {
+    FRONT_END_GENERAL_CHECK(place->is_input(), "Setting element type is supported for input only.");
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
+    auto it = m_descriptors.find(pytorch_place->get_tensor_index());
+    if (it != m_descriptors.end()) {
+        it->second.m_type = type;
+    }
+}
+
+ov::element::Type InputModel::get_element_type(const Place::Ptr& place) const {
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
+    auto it = m_descriptors.find(pytorch_place->get_tensor_index());
+    if (it != m_descriptors.end()) {
+        return it->second.m_type;
+    }
+    return element::dynamic;
+}
+
+void InputModel::set_tensor_value(const Place::Ptr& place, const void* value) {
+    FRONT_END_GENERAL_CHECK(place->is_input(), "Setting value is supported for input only.");
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
+    auto it = m_descriptors.find(pytorch_place->get_tensor_index());
+    if (it != m_descriptors.end()) {
+        auto el_type = it->second.m_type;
+        auto p_shape = it->second.m_pshape;
+        FRONT_END_GENERAL_CHECK(el_type.is_static() && p_shape.is_static(),
+                                "Shape and type must be statically defined before calling set_tensor_value");
```

Comment:
Is it possible to deduce type from the model without requesting it?

Reply:
Are you talking about reverse infer? We do not do any conversion on this point, so we cannot deduce it now, it is possible to do it after conversion, but it is not guaranteed and it seems to be out of scope of this API. If the model is traced, then type and shape will exist in descriptors map, so in that case everything is expected, type will also be known if it was annotated in initial model even if scripting is used.